code/modules/mob/living/basic/minebots/minebot.dm

```diff
@@ -0,0 +1,179 @@
+	desc = "The instructions printed on the side read: This is a small robot used to support miners, can be set to search and collect loose ore, or to help fend off wildlife."
+	gender = NEUTER
+	icon = 'icons/mob/silicon/aibots.dmi'
+	icon_state = "mining_drone"
+	icon_living = "mining_drone"
+	basic_mob_flags = DEL_ON_DEATH
+	status_flags = CANSTUN|CANKNOCKDOWN|CANPUSH
+	mouse_opacity = MOUSE_OPACITY_ICON
+	combat_mode = TRUE
+	habitable_atmos  = list("min_oxy" = 0, "max_oxy" = 0, "min_plas" = 0, "max_plas" = 0, "min_co2" = 0, "max_co2" = 0, "min_n2" = 0, "max_n2" = 0)
+	minimum_survivable_temperature = 0
+	health = 125
+	maxHealth = 125
+	melee_damage_lower = 15
+	melee_damage_upper = 15
+	obj_damage = 10
+	attack_verb_continuous = "drills"
+	attack_verb_simple = "drill"
+	attack_sound = 'sound/weapons/circsawhit.ogg'
+	sentience_type = SENTIENCE_MINEBOT
+	speak_emote = list("states")
+	mob_biotypes = MOB_ROBOTIC
+	death_message = "blows apart!"
+	light_system = MOVABLE_LIGHT
+	light_range = 6
+	light_on = FALSE
+	combat_mode = FALSE
+	ai_controller = /datum/ai_controller/basic_controller/minebot
+	///the access card we use to access mining
+	var/obj/item/card/id/access_card
+	///the gun we use to kill
+	var/obj/item/gun/energy/recharge/kinetic_accelerator/minebot/stored_gun
+	///the commands our owner can give us
+	var/list/pet_commands = list(
+		/datum/pet_command/idle,
+		/datum/pet_command/minebot_ability/light,
+		/datum/pet_command/minebot_ability/dump,
+		/datum/pet_command/automate_mining,
+		/datum/pet_command/free/minebot,
+		/datum/pet_command/follow,
+		/datum/pet_command/point_targetting/attack/minebot,
+	)
+
+/mob/living/basic/mining_drone/Initialize(mapload)
+	. = ..()
+
+	var/static/list/death_loot = list(/obj/effect/decal/cleanable/robot_debris)
+	AddElement(/datum/element/death_drops, death_loot)
+	AddElement(/datum/element/footstep, FOOTSTEP_OBJ_ROBOT, 1, -6, sound_vary = TRUE)
+	AddComponent(\
+		/datum/component/tameable,\
+		food_types = list(/obj/item/stack/ore),\
+		tame_chance = 100,\
+		bonus_tame_chance = 5,\
+		after_tame = CALLBACK(src, PROC_REF(activate_bot)),\
+	)
+
+	var/datum/action/cooldown/mob_cooldown/minedrone/toggle_light/toggle_light_action = new(src)
+	var/datum/action/cooldown/mob_cooldown/minedrone/toggle_meson_vision/toggle_meson_vision_action = new(src)
+	var/datum/action/cooldown/mob_cooldown/minedrone/dump_ore/dump_ore_action = new(src)
+	toggle_light_action.Grant(src)
+	toggle_meson_vision_action.Grant(src)
+	dump_ore_action.Grant(src)
+	ai_controller.set_blackboard_key(BB_MINEBOT_LIGHT_ABILITY, toggle_light_action)
+	ai_controller.set_blackboard_key(BB_MINEBOT_DUMP_ABILITY, dump_ore_action)
+
+	stored_gun = new(src)
+	var/obj/item/implant/radio/mining/imp = new(src)
+	imp.implant(src)
+	access_card = new /obj/item/card/id/advanced/gold(src)
+	SSid_access.apply_trim_to_card(access_card, /datum/id_trim/job/shaft_miner)
+
+	RegisterSignal(src, COMSIG_MOB_TRIED_ACCESS, PROC_REF(attempt_access))
```

Comment:
we really need to elementize this since i know regular drones do something similar and it feels icky

Reply:
access card is a variable on simplemobs already